### PR TITLE
adds legacy contact-info api to the new one

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Working towards migrating to the new contact-info, the commit expands api parity between the legacy and the new one.

#### Summary of Changes
Added legacy contact-info api to the new one.

Part of https://github.com/anza-xyz/agave/pull/823 made into a separate commit.

### CODE REVIEW

File: gossip/src/contact_info.rs [hunk 1]
```diff
@@ -366,6 +366,29 @@ impl ContactInfo {
...
 
+    /// New random ContactInfo for tests and simulations.
+    pub fn new_rand<R: rand::Rng>(rng: &mut R, pubkey: Option<Pubkey>) -> Self {
+        let delay = 10 * 60 * 1000; // 10 minutes
+        let now = solana_sdk::timing::timestamp() - delay + rng.gen_range(0..2 * delay);
+        let pubkey = pubkey.unwrap_or_else(solana_sdk::pubkey::new_rand);
+        let mut node = ContactInfo::new_localhost(&pubkey, now);
+        let _ = node.set_gossip((Ipv4Addr::LOCALHOST, rng.gen_range(1024..u16::MAX)));
+        node
+    }
```
Comment (behzadnouri) [inline gossip/src/contact_info.rs:377]: equivalent to:
https://github.com/anza-xyz/agave/blob/fb3bf5534/gossip/src/legacy_contact_info.rs#L157-L165

File: gossip/src/contact_info.rs [hunk 2]
```diff
@@ -366,6 +366,29 @@ impl ContactInfo {
...
+        let mut node = Self::new(
+            Pubkey::default(),
+            solana_sdk::timing::timestamp(), // wallclock
+            0,                               // shred_version
+        );
+        if let Err(err) = node.set_gossip(*gossip_addr) {
+            error!("Invalid entrypoint: {gossip_addr}, {err:?}");
+        }
+        node
+    }
```
Comment (behzadnouri) [inline gossip/src/contact_info.rs:390]: equivalent to:
https://github.com/anza-xyz/agave/blob/fb3bf5534/gossip/src/legacy_contact_info.rs#L167-L175

File: gossip/src/contact_info.rs [hunk 3]
```diff
@@ -409,6 +432,16 @@ impl ContactInfo {
...
 
+impl Default for ContactInfo {
+    fn default() -> Self {
+        Self::new(
+            Pubkey::default(),
+            0, // wallclock
+            0, // shred_version
+        )
+    }
+}
```
Comment (behzadnouri) [inline gossip/src/contact_info.rs:443]: equivalent to:
https://github.com/anza-xyz/agave/blob/fb3bf5534/gossip/src/legacy_contact_info.rs#L118-L136

**Review by gregcusack — APPROVED** [review]
lgtm!

**Pull request merged**